packages/observablehq-compiler/vite.config.ts

```diff
@@ -1,20 +1,118 @@
+import { dirname, resolve } from "node:path";
+import { fileURLToPath } from "node:url";
+
+import { defineConfig } from "vite";
+import topLevelAwait from "vite-plugin-top-level-await";
 import { createHpccViteConfig, browserConfig, nodeConfig } from "@hpcc-js/esbuild-plugins";
+
 import pkg from "./package.json" with { type: "json" };
 
-const myBrowserConfig = { ...browserConfig };
-myBrowserConfig.optimizeDeps = {
-    include: ["acorn", "acorn-walk", "@observablehq/parser"]
-};
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
 
-const myNodeConfig = { ...nodeConfig };
-myNodeConfig.optimizeDeps = {
-    include: ["acorn", "acorn-walk", "@observablehq/parser"]
-};
+interface NpmSpecifier {
+    /** a package name, such as "d3" */
+    name: string;
+    /** at sign and semver range, such as "@7", or the empty string */
+    range: string;
+    /** slash and path, such as "/foo", or the empty string */
+    path: string;
+}
 
-export default createHpccViteConfig(pkg, {
-    configOverrides: {
-        test: {
-            projects: [myBrowserConfig, myNodeConfig]
-        }
+/** Note: specifier here does not start with `npm:`. */
+function parseNpmSpecifier(specifier: string): NpmSpecifier {
+    const parts = specifier.split("/");
+    const namerange = specifier.startsWith("@")
+        ? [parts.shift()!, parts.shift()!].join("/")
+        : parts.shift()!;
+    const ranged = namerange.indexOf("@", 1);
+    const name = ranged > 0 ? namerange.slice(0, ranged) : namerange;
+    const range = ranged > 0 ? namerange.slice(ranged) : "";
+    const path = parts.length > 0 ? `/${parts.join("/")}` : "";
+    return { name, range, path };
+}
+
+function getDefaultRange(name: string): string {
+    switch (name) {
+        case "@duckdb/duckdb-wasm":
+            return "@1.29.0"; // https://github.com/duckdb/duckdb-wasm/issues/1994
+        case "apache-arrow":
+            return "@17.0.0"; // to match @duckdb/duckdb-wasm 1.29.0
+        default:
+            return "";
     }
-});
+}
+
+function getDefaultPath(name: string): string {
+    switch (name) {
+        case "mermaid":
+            return "/dist/mermaid.esm.min.mjs/+esm";
+        case "echarts":
+            return "/dist/echarts.esm.min.js/+esm";
+        case "jquery-ui":
+            return "/dist/jquery-ui.js/+esm";
```

Comment:
Are these all here to support different chart libraries? This one just jumped out at me because there's not jQuery (UI or otherwise) anywhere in this PR or @hpcc-js is there?